plugins/utils/which-key.nix

```diff
@@ -121,153 +475,44 @@ with lib;
         };
       in
       {
-        border = helpers.defaultNullOpts.mkBorder "none" "which-key" "";
-        position = helpers.defaultNullOpts.mkEnumFirstDefault [
+        border = defaultNullOpts.mkBorder "none" "which-key" "";
+        position = defaultNullOpts.mkEnumFirstDefault [
           "bottom"
           "top"
         ] "";
-        margin = helpers.defaultNullOpts.mkNullable spacingOptions {
+        margin = defaultNullOpts.mkNullable spacingOptions {
           top = 1;
           right = 0;
           bottom = 1;
           left = 0;
         } "extra window margin";
-        padding = helpers.defaultNullOpts.mkNullable spacingOptions {
+        padding = defaultNullOpts.mkNullable spacingOptions {
           top = 1;
           right = 2;
           bottom = 1;
           left = 2;
         } "extra window padding";
-        winblend = helpers.defaultNullOpts.mkNullable (types.ints.between 0
+        winblend = defaultNullOpts.mkNullable (types.ints.between 0
           100
         ) 0 "0 for fully opaque and 100 for fully transparent";
       };
```

Comment:
These options continuing to be `defaultNullOpts` is what's leading to weirdness in https://github.com/nix-community/nixvim/pull/1990#discussion_r1708188447 and https://github.com/nix-community/nixvim/pull/1990#discussion_r1708170104.

In order for `options.*.isDefined` to be useful, the option must **not** have a default value. If it has a default, `isDefined` is _always_ true.

Additionally, we also want these options to be `visible = false`, since they are deprecated.

We can achieve this by using `mkOption` directly, we just have to manually replicate whatever type the `defaultNullOpts` helper would have produced. Luckily, since the option will not be visible, there's not much point in setting a plugin-default anymore (which would be awkward when using `mkOption` directly).

Here's an example for one of the options:

```nix
position = mkOption {
  type = nullOr (maybeRaw (enumOf [
    "bottom"
    "top"
  ]));
  description = ''
    This option is deprecated, you should use `plugins.which-key.settings.win` instead.
  '';
  visible = false;
  # Note: no default is defined
}
```